tools/crates-validator/src/lib.rs

```diff
@@ -218,37 +259,34 @@ impl Validator {
         cmd.env_clear();
         cmd.envs(&filtered_env);
 
-        if profile.std {
+        if profile.settings.std {
             cmd.arg("--std");
         }
 
-        match (&self.context.risc0_path, &self.context.risc0_gh_branch) {
-            (None, None) => {
-                bail!("Profile must assign either 'risc0_template_path' or 'risc0_gh_branch'")
+        match self.repo() {
+            Repo::Tag(_) => {
+                cmd.arg("--tag");
+                cmd.arg(self.repo().value());
             }
-            (Some(..), Some(..)) => bail!(
-                "Profile assigns both 'risc0_template_path' and 'risc0_gh_branch' pick just one"
-            ),
-            (Some(r0_path), None) => {
-                let r0_template_path = Path::new(r0_path).join("templates").join("rust-starter");
-                if !r0_template_path.exists() {
-                    bail!(
-                        "Failed to find {} on disk",
-                        r0_template_path.to_string_lossy()
-                    );
+            Repo::Branch(_) => {
+                cmd.arg("--branch");
+                cmd.arg(self.repo().value());
+            }
```

Comment:
Neither of those branches seems to result in working compilation steps, I can only execute the validation successfully using the path. Any idea why?

Reply:
@mothran , maybe you'd have a better idea on why?

Reply:
Do you have some errors I could look at?